apps/mobile/src/components/PostCard/PostBodyProposal/PostBodyProposal.js

```diff
@@ -96,19 +100,19 @@ export default function PostBodyProposal ({
   function handleVote (optionId) {
     if (votingMethod === VOTING_METHOD_SINGLE) {
       if (currentUserVotesOptionIds.includes(optionId)) {
-        dispatch(removeProposalVote({ optionId, postId: id }))
+        removeProposalVote({ optionId, postId: id })
       } else if (currentUserVotesOptionIds.length === 0) {
-        dispatch(addProposalVote({ optionId, postId: id }))
+        addProposalVote({ optionId, postId: id })
```

Comment:
You can just `const { fetching: addProposalVoteMutationFetching }` or above in `const [{ fetching: addProposalVoteMutationFetching, addProposalVote] = useMutation(addProposalVoteMutation)`, as you wish. I would recommend a `const loading = useMemo(() => addProposalVoteMutationFetching && addProposalVoteMutationFetching && addProposalVoteMutationFetching, [addProposalVoteMutationFetching, addProposalVoteMutationFetching, addProposalVoteMutationFetching]` or similar and then using that loading state in the component to block actions until they're resolved. Even with an updater or optimistic updater that reconciles to the cache without and invalidate, it can still take a visual period of time so best to handle the UI blocking part directly in the component here.